evm/src/witness/traces.rs

```diff
@@ -76,6 +84,12 @@ impl<T: Copy> Traces<T> {
                 .iter()
                 .map(|op| op.input.len() / keccak_sponge::columns::KECCAK_RATE_BYTES + 1)
                 .sum(),
+            poseidon_len: self.poseidon_inputs.len(),
+            poseidon_sponge_len: self
+                .poseidon_sponge_ops
+                .iter()
+                .map(|op| op.input.len() / poseidon::columns::POSEIDON_SPONGE_RATE + 1)
```

Comment:
nit: we may need to adjust this, as we will *mostly* use it for compression within SMTs, with something like `if op.input.len() == SPONGE_RATE { 1 } else { op.input.len() / poseidon::columns::POSEIDON_SPONGE_RATE + 1 }`